### PR TITLE
Adding quality Check endpoints

### PR DESCRIPTION
**Pull request by ReddytoRumble**

### CODE REVIEW

**Comment by tabladrum** [comment]
Can you squash the commits

File: api-audit/src/main/java/com/capitalone/dashboard/rest/AuditController.java [hunk 1]
```diff
@@ -54,12 +62,13 @@ public AuditController(AuditService auditService) {
      * @param request
      * @return
      */
-    @RequestMapping(value = "/peerReview", method = GET, produces = APPLICATION_JSON_VALUE)
+    @RequestMapping(value = {"/peerReview","/peerReview/testCode"}, method = GET, produces = APPLICATION_JSON_VALUE)
```
Comment (tabladrum) [inline api-audit/src/main/java/com/capitalone/dashboard/rest/AuditController.java]: why is peerReview/testCode in there?

File: api-audit/src/main/java/com/capitalone/dashboard/rest/AuditController.java [hunk 2]
```diff
@@ -74,5 +83,70 @@ public AuditController(AuditService auditService) {
...
+     * @param request
+     * @return
+     */
+    @RequestMapping(value = "/staticCodeAnalysis/businessApp", method = GET, produces = APPLICATION_JSON_VALUE)
+    public ResponseEntity<List<StaticAnalysisResponse>> staticCodeAnalysis(StaticAnalysisRequest request) throws HygieiaException {
+        
+        List<StaticAnalysisResponse> staticAnalysisResponse;
+        try {
+            staticAnalysisResponse = auditService.getCodeQualityAudit(request.getComponent(),request.getArtifactGroup(),request.getArtifactName(),request.getArtifactVersion());
+        } catch (Exception e) {
```
Comment (tabladrum) [inline api-audit/src/main/java/com/capitalone/dashboard/rest/AuditController.java]: We want to stay away from catching Exception. Need to catch specific exceptions.

File: api-audit/src/main/java/com/capitalone/dashboard/rest/AuditController.java [hunk 3]
```diff
@@ -74,5 +83,70 @@ public AuditController(AuditService auditService) {
...
+            @RequestParam(value = "artVersion", required = true) String artVersion) throws HygieiaException {
+
+        List<StaticAnalysisResponse> staticAnalysisResponse;
+        try {
+            staticAnalysisResponse = auditService.getCodeQualityAudit(artGroup, artName, artVersion);
+        } catch (IOException e) {
+            throw new HygieiaException("Can not retrieve data for the requested artifact.", HygieiaException.BAD_DATA);
+        }
+        return ResponseEntity.ok().body(staticAnalysisResponse);
+    }
```
Comment (tabladrum) [inline api-audit/src/main/java/com/capitalone/dashboard/rest/AuditController.java]: May be the actual implementations should throw HygieiaException as standards?

**Comment by ReddytoRumble** [comment]
@tabladrum Updated to incorporate review changes.

File: api-audit/src/test/java/com/capitalone/dashboard/config/TestConfig.java [hunk 4]
```diff
@@ -15,4 +15,5 @@
     public AuditService auditService() {
         return Mockito.mock(AuditService.class);
     }
+
```
Comment (AuditHawks) [inline api-audit/src/test/java/com/capitalone/dashboard/config/TestConfig.java]: @ReddytoRumble @tabladrum Please remove this file from the PR if it doesn't have any changes.

Comment (satishc1) [inline api-audit/src/test/java/com/capitalone/dashboard/config/TestConfig.java]: @ReddytoRumble @tabladrum Please remove this file from the PR if there are no changes.

File: api-audit/src/main/java/com/capitalone/dashboard/service/AuditServiceImpl.java [hunk 5]
```diff
@@ -332,4 +364,265 @@ public JobReviewResponse getBuildJobReviewResponse(String jobUrl, String jobName
...
+     * @param businessApp Business Application
+     * @param artifactGroup Artifact Group
+     * @param artifactName Artifact Name
+     * @param artifactVersion Artifact Version
+     * @return List of StaticAnalysisResponse
+     * @throws HygieiaException 
+     */
+    @Override
+    public List<StaticAnalysisResponse> getCodeQualityAudit(String businessApp,String artifactGroup, String artifactName,String artifactVersion) throws HygieiaException {
+
```
Comment (satishc1) [inline api-audit/src/main/java/com/capitalone/dashboard/service/AuditServiceImpl.java]: @ReddytoRumble @tabladrum Why is BAP an input here? We should either query based on artifact parameters or from the dashboardReview by checking the related collectorItem.

File: api-audit/src/main/java/com/capitalone/dashboard/service/AuditServiceImpl.java [hunk 6]
```diff
@@ -332,4 +364,265 @@ public JobReviewResponse getBuildJobReviewResponse(String jobUrl, String jobName
...
+     * @param businessApp Business Application
+     * @param artifactGroup Artifact Group
+     * @param artifactName Artifact Name
+     * @param artifactVersion Artifact Version
+     * @return List of StaticAnalysisResponse
+     * @throws HygieiaException 
+     */
+    
+    public TestResultsResponse getTestResultExecutionDetails(String businessApp,String artifactGroup, String artifactName,String artifactVersion) throws HygieiaException {
+    	Dashboard dashboard = this.getDashboardDetails(businessApp);
```
Comment (satishc1) [inline api-audit/src/main/java/com/capitalone/dashboard/service/AuditServiceImpl.java]: @ReddytoRumble @tabladrum Why is BAP an input here? We should either query it based on artifact parameters or from the dashboardReview using related collectorItem.

File: api-audit/src/main/java/com/capitalone/dashboard/service/AuditServiceImpl.java [hunk 7]
```diff
@@ -332,4 +364,265 @@ public JobReviewResponse getBuildJobReviewResponse(String jobUrl, String jobName
...
+        
+        
+		return testResultsResponse;
+        
+    }
+    
+    /**
+     * Retrieves code quality profile changeset for a given time period and determines if change author matches commit author within time period
+     * @param businessApp Business Application
+     * @param artifactGroup Artifact Group
```
Comment (satishc1) [inline api-audit/src/main/java/com/capitalone/dashboard/service/AuditServiceImpl.java]: @ReddytoRumble @tabladrum BAP here as well ? Do we need it ?

File: api-audit/src/main/java/com/capitalone/dashboard/service/AuditServiceImpl.java [hunk 8]
```diff
@@ -332,4 +364,265 @@ public JobReviewResponse getBuildJobReviewResponse(String jobUrl, String jobName
...
+	        codeQualityGateValidationResponse.setQualityGateChangePerformers(qualityProfileChangePerformers);
+	    }
+        codeQualityGateValidationResponse.setCommitAuthors(authors);
+        return codeQualityGateValidationResponse;
+    }
+    
+    
+    
+    private Dashboard getDashboardDetails(String businessApp) throws HygieiaException {
+    	Cmdb cmdbItem = cmdbRepository.findByConfigurationItemIgnoreCase(businessApp);
```
Comment (satishc1) [inline api-audit/src/main/java/com/capitalone/dashboard/service/AuditServiceImpl.java]: @ReddytoRumble @tabladrum Why is this method needed ?  We should either audit an individual widget or a Dashboard. If you are auditing an individual widget, why is the dashboard coming in the picture ?

File: api-audit/src/main/java/com/capitalone/dashboard/service/AuditServiceImpl.java [hunk 9]
```diff
@@ -15,42 +18,65 @@
...
+import java.util.HashMap;
 import java.util.HashSet;
 import java.util.Iterator;
 import java.util.List;
 import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
 
 @Component
 public class AuditServiceImpl implements AuditService {
```
Comment (satishc1) [inline api-audit/src/main/java/com/capitalone/dashboard/service/AuditServiceImpl.java]: @ReddytoRumble @tabladrum The dashboardReview needs to be modified to include the gates that you audited individually.

File: api-audit/src/main/java/com/capitalone/dashboard/service/AuditServiceImpl.java [hunk 10]
```diff
@@ -332,4 +364,265 @@ public JobReviewResponse getBuildJobReviewResponse(String jobUrl, String jobName
...
+    }
+    
+    /**
+     * Helper method to get a list of collector items for a componentId
+     * @param componentId
+     * @return List of collector items
+     */
+
+    private List<CollectorItem> getCollectorItems(ObjectId componentId) {
+        com.capitalone.dashboard.model.Component component = componentRepository.findOne(componentId);
```
Comment (satishc1) [inline api-audit/src/main/java/com/capitalone/dashboard/service/AuditServiceImpl.java]: @ReddytoRumble @tabladrum Same as above method.. why is it needed ?

**Pull request closed without merging**